### ISSUE TITLE
Translating examples and protocols to 2.8

### ISSUE THREAD

**Issue by arekbulski** [construct/construct#158]
Code needs a translation, quite a lot of it. Classes changed and names moved to `name / Struct(field1...)`.
http://stackoverflow.com/questions/39634228/need-regex-to-fix-construct-2-8-name-less

**Comment by arekbulski** [comment]
Bug in bitmap building, looks like flagbuildnone is not inherited?
```py
self = <Mapping: None +nonbuild>, obj = None, context = Container(signature='BM')(file_size=16430)(data_offset=1078)(header_size=40)

    def _encode(self, obj, context):
        try:
            return self.encoding[obj]
        except (KeyError, TypeError):
            if self.encdefault is NotImplemented:
>               raise MappingError("no encoding mapping for %r" % (obj,))
E               MappingError: no encoding mapping for None
```

**Comment by arekbulski** [comment]
PNG also fails when building.

**Comment by arekbulski** [comment]
Merged with #173
